[example] Explain package choice (#23938)
mui-org/material-ui: examples/gatsby/README.md

## The idea behind the example

> [Gatsby](https://github.com/gatsbyjs/gatsby) is a static site generator for React. The project template adds `@material-ui/core` and it's peer dependencies.

Reviewer: ```suggestion
[Gatsby](https://github.com/gatsbyjs/gatsby) is a static site generator for React.
Additionally, it adds `@material-ui/core` and it's peer dependencies.
```
